protobuf-kafkaconnect-converter/src/test/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/ConnectSchemaToProtobufSchemaConverterTest.java

```diff
@@ -0,0 +1,158 @@
+        return ImmutableMap.<String, Schema>builder()
+            .put("nonNumberTag", new SchemaBuilder(Schema.Type.INT16).parameter(PROTOBUF_TAG, "jsf").build())
+            .put("nullNumberTag", new SchemaBuilder(Schema.Type.INT16).parameter(PROTOBUF_TAG, null).build())
+            .put("invalidInt32Metadata", new SchemaBuilder(Schema.Type.INT32).parameter(PROTOBUF_TYPE, "int64").build())
+            .put("invalidInt64Metadata",
+                new SchemaBuilder(Schema.Type.INT32).parameter(PROTOBUF_TYPE, "string").build())
+            .build();
+    }
+
+    private static Stream<Arguments> getConnectSchemaTestCases() {
+        return Stream.of(
+            Arguments.of(
+                "PrimitiveTypes",
+                getPrimitiveTypes(),
+                "PrimitiveProtobufSchema.filedescproto"
+            )
+        );
+    }
+
+    private static Stream<Arguments> getConnectSchemaExceptionTestCases() {
+        return getPrimitiveTypesForExceptions()
+            .entrySet()
+            .stream()
+            .map(entry ->
+                Arguments.of(entry.getKey(), ImmutableMap.of(entry.getKey(), entry.getValue()))
+            );
+    }
+
+    @ParameterizedTest(name = "{index} {0}")
+    @MethodSource("getConnectSchemaTestCases")
+    public void fromConnectSchema_convertsConnectSchemaToProtobufSchema(String fileName,
+        Map<String, Schema> connectTypes, String expectedProtobufSchemaFile)
+        throws IOException {
+        final SchemaBuilder parentSchemaBuilder = new SchemaBuilder(Schema.Type.STRUCT);
+        parentSchemaBuilder.name(fileName);
+
+        connectTypes
+            .forEach(parentSchemaBuilder::field);
+
+        final Schema connectSchema = parentSchemaBuilder.build();
+
+        final Descriptors.FileDescriptor protobufSchema =
+            CONNECT_SCHEMA_TO_PROTOBUF_SCHEMA_CONVERTER.convert(connectSchema);
+
+        final String actualSchema = protobufSchema.toProto().toString();
+
+        final String expectedSchema =
+            new String(Files.readAllBytes(Paths.get(TEST_RESOURCE_PATH, expectedProtobufSchemaFile)),
+                Charset.defaultCharset());
+
+        assertEquals(expectedSchema, actualSchema);
+    }
+
+    @ParameterizedTest(name = "{index} {0}")
+    @MethodSource("getConnectSchemaExceptionTestCases")
+    public void fromConnectSchema_convertsConnectSchemaToProtobufSchema_forExceptions(String fileName,
+        Map<String, Schema> connectTypes) {
+        final SchemaBuilder parentSchemaBuilder = new SchemaBuilder(Schema.Type.STRUCT);
+        parentSchemaBuilder.name(fileName);
+
+        connectTypes
+            .forEach(parentSchemaBuilder::field);
+
+        final Schema connectSchema = parentSchemaBuilder.build();
+
+        assertThrows(IllegalArgumentException.class,
+            () -> CONNECT_SCHEMA_TO_PROTOBUF_SCHEMA_CONVERTER.convert(connectSchema));
+    }
+
+    @Test
+    public void fromConnectSchema_onNullSchema_ThrowsException() {
```

Comment:
ARe all data types covered in these tests ? Unions ?

Reply:
This is for just the Primitive types. We will add other types.